File: entities/ocr_result.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type DocumentOCRResult struct {
	ID               uuid.UUID `gorm:"primaryKey" json:"id"`
	DocumentID       uuid.UUID `gorm:"not null;index" json:"document_id"`
	Provider         string    `gorm:"size:50;not null" json:"provider"`
	RawJSON          string    `gorm:"type:text;not null" json:"raw_json"`
	ParsedJSON       string    `gorm:"type:text;not null" json:"parsed_json"`
	Confidence       float64   `gorm:"not null" json:"confidence"`
	ProcessingTimeMS int64     `gorm:"not null" json:"processing_time_ms"`
	CreatedAt        time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (o *DocumentOCRResult) BeforeCreate(tx *gorm.DB) (err error) {
	if o.ID == uuid.Nil {
		o.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/ocr_result.py
import uuid
from sqlalchemy import Column, String, Integer, BigInteger, Float, DateTime, Text, ForeignKey
from sqlalchemy.sql import func
from database import Base

class DocumentOCRResult(Base):
    __tablename__ = "document_ocr_results"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    document_id = Column(String(36), ForeignKey("doctor_documents.document_id", ondelete="CASCADE"), nullable=False, index=True)
    provider = Column(String(50), nullable=False)
    raw_json = Column(Text, nullable=False)
    parsed_json = Column(Text, nullable=False)
    confidence = Column(Float, nullable=False)
    processing_time_ms = Column(BigInteger, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
